src/app/qgsclipboard.cpp

```diff
@@ -110,38 +117,50 @@ QString QgsClipboard::generateClipboardText() const
         if ( format == AttributesWithWKT )
         {
           if ( it->hasGeometry() )
-            textFields += it->geometry().asWkt();
+          {
+            QString wkt = it->geometry().asWkt();
+            textFields += wkt;
+            htmlFields += QStringLiteral( "<td>%1</td>" ).arg( wkt );
+          }
           else
           {
             textFields += QgsApplication::nullRepresentation();
+            htmlFields += QStringLiteral( "<td>%1</td>" ).arg( QgsApplication::nullRepresentation() );
           }
         }
 
         for ( int idx = 0; idx < attributes.count(); ++idx )
         {
-          // QgsDebugMsg(QString("inspecting field '%1'.").arg(it2->toString()));
-          if ( attributes.at( idx ).toString().contains( QStringLiteral( "\n" ), Qt::CaseInsensitive ) )
-            textFields += QStringLiteral( "\"" ) + attributes.at( idx ).toString() + QStringLiteral( "\"" );
+          QString value = attributes.at( idx ).toString();
+          if ( attributes.at( idx ).toString().contains( QStringLiteral( "\n" ), Qt::CaseInsensitive ) ||
```

Comment:
```suggestion
          if ( attributes.at( idx ).toString().contains( '\n', Qt::CaseInsensitive ) ||
```

Reply:
@nyalldawson , actually, the case insensitive flag can also go, wouhou.